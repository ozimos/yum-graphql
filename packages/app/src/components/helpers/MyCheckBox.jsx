import { withFormsy, propTypes } from 'formsy-react';
import React from 'react';

class CheckBox extends React.Component {
  constructor(props) {
    super(props);
    this.changeValue = this.changeValue.bind(this);
  }

  changeValue(event) {

    this.props.setValue(event.target.checked);
  }

  render() {

    return (
      <React.Fragment>

        <input
          style={{ visibility: 'hidden', margin: '-5px' }}
          onChange={this.changeValue}
          type="checkbox"
          value={this.props.getValue() || false}
          id="isCaterer"
          name="isCaterer"
        />

      </React.Fragment>
    );
  }
}

CheckBox.propTypes = {
  ...propTypes
};

export { CheckBox };

export default withFormsy(CheckBox);
