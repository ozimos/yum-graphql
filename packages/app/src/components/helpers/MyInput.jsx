import { withFormsy, propTypes } from 'formsy-react';
import PropTypes from 'prop-types';
import React from 'react';

class Input extends React.Component {

  constructor(props) {
    super(props);
    this.changeValue = this.changeValue.bind(this);
  }

  changeValue(event) {

    this.props.setValue(event.currentTarget.value);
  }

  render() {
    const errorMessage = this.props.getErrorMessage();

    return (

      <React.Fragment>

        <input
          className="formsy-input"
          style={this.props.style}
          onChange={this.changeValue}
          type={this.props.typeOfInput}
          value={this.props.getValue() || ''}
          placeholder={this.props.placeholder}
          maxLength="50"
          max="99999"
        />

        <div
          className="error"
          style={{
            color: 'red', fontSize: '1rem'
          }}
        >
          {errorMessage}
        </div>

      </React.Fragment>
    );
  }
}

Input.defaultProps = {
  placeholder: '',
};

Input.propTypes = {
  typeOfInput: PropTypes.string.isRequired,
  placeholder: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  ...propTypes
};

export { Input };

export default withFormsy(Input);
