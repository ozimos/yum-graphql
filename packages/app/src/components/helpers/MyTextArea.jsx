import { withFormsy, propTypes } from 'formsy-react';
import PropTypes from 'prop-types';
import React from 'react';

class TextArea extends React.Component {
  constructor(props) {
    super(props);
    this.changeValue = this.changeValue.bind(this);
  }

  changeValue(event) {

    this.props.setValue(event.currentTarget.value);
  }

  render() {
    const errorMessage = this.props.getErrorMessage();

    return (
      <React.Fragment>
        <textarea
          onChange={this.changeValue}
          value={this.props.getValue() || this.props.initialValue}
          placeholder={this.props.placeholder}
          maxLength={100}
        />
        <div
          style={{
            color: 'red', fontSize: '1rem'
          }}
        >
          {errorMessage}
        </div>
      </React.Fragment>
    );
  }
}

TextArea.defaultProps = {
  placeholder: ''
};

TextArea.propTypes = {
  placeholder: PropTypes.string,
  ...propTypes
};

export { TextArea };

export default withFormsy(TextArea);
