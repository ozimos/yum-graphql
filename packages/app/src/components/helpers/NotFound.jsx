import React from 'react';
import { Link } from 'react-router-dom';
import '../../../public/styles/dashboard.scss';


import ConnectedNav from '../nav/ConnectedNav';


const NotFound = () => (
  <div id="not-found" className="canvas2">
    <div className="container3">
      <header className="header extra">
        <ConnectedNav />
      </header>
      <div>
        <h3> Sorry, the page you are looking for does not exist</h3>
        <h4> <Link to="/">Go Home</Link></h4>
      </div>
    </div>
  </div>
);

export default NotFound;

