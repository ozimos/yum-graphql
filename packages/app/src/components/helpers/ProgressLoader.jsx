import React from 'react';
import PropTypes from 'prop-types';

import LinearProgress from '@material-ui/core/LinearProgress';

const ProgressLoader = props =>
  (
    <div className="progression">
      <LinearProgress
        style={{ height: '10px' }}
        variant="determinate"
        value={props.upload}
      />
    </div>
  );

ProgressLoader.propTypes = {
  upload: PropTypes.number.isRequired,
};

export default ProgressLoader;

