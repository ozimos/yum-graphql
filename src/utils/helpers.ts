import { sign } from 'jsonwebtoken';
import { TOKEN_PASSWORD, tokens } from './constants';

export const handleError = (error) => {
    // add any other logging mechanism here e.g. Sentry
    return error;
};
interface CreateTokenInput {
    id: string;
    role: string;
}
export const generateAccessToken = ({ id: userId, role }: CreateTokenInput) => {
    const accessToken = sign(
        {
            userId,
            role,
            type: tokens.access.name,
            timestamp: Date.now(),
        },
        TOKEN_PASSWORD,
        {
            expiresIn: tokens.access.expiry,
        },
    );
    return accessToken;
};
