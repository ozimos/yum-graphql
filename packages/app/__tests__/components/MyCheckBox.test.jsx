/* global React:false, shallow:false toJson:false */

import { CheckBox } from '../../src/components/helpers/MyCheckBox';

describe('MyCheckBox Component', () => {

  const props = {
    getValue: jest.fn(),
    setValue: jest.fn(),
    getErrorMessage: jest.fn(),
    typeOfInput: 'checkbox',
    style: '',
  };

  const setup = () => shallow(<CheckBox {...props} />);

  it('should render without throwing an error', () => {
    const wrapper = setup();

    expect(wrapper
      .exists(<input type="checkbox" />)).toBe(true);
  });

  it('renders correctly', () => {
    const wrapper = setup();

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call `changeValue` when the input is changed', () => {
    const wrapper = setup();

    const changeValueSpy = jest.spyOn(wrapper.instance(), 'changeValue');
    wrapper.instance().changeValue({ target: { checked: true } });
    expect(changeValueSpy).toHaveBeenCalled();
  });
});

