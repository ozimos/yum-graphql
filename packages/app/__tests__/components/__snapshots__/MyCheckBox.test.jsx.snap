// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCheckBox Component renders correctly 1`] = `
<React.Fragment>
  <input
    id="isCaterer"
    name="isCaterer"
    onChange={[Function]}
    style={
      Object {
        "margin": "-5px",
        "visibility": "hidden",
      }
    }
    type="checkbox"
    value={false}
  />
</React.Fragment>
`;
